shm/source/Item.cpp

```diff
@@ -0,0 +1,35 @@
+#include "Item.hpp"
+
+Item::Item(const std::string& name, const size_t amount, const size_t basePrice, const Rarity rarity):
+    Cargo(name, amount, basePrice),
+    rarity_(rarity)
+{}
+
+size_t Item::getPrice() const {
+    switch (rarity_) {
+    case Rarity::Common:
+        return getBasePrice() / 3;
+
+    case Rarity::Rare:
+        return getBasePrice() / 2;
+
+    case Rarity::Epic:
+        return getBasePrice();
+
+    case Rarity::Legendary:
+        return getBasePrice() * 4;
+    }
```

Comment:
better to always multiply. So common should be 1, rare eg. 3 etc.. Dividing always makes trouble in testing, because you lose precision.